velox/type/Date.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/common/base/Exceptions.h"
+#include "velox/type/StringView.h"
+
+#include <iomanip>
+#include <sstream>
+#include <string>
+
+#include <folly/dynamic.h>
+
+namespace facebook::velox {
+
+struct Date {
+ public:
+  constexpr Date() : days_(0) {}
+  constexpr Date(int32_t days) : days_(days) {}
+
+  int32_t days() const {
+    return days_;
+  }
+
+  bool operator==(const Date& other) const {
+    return days_ == other.days_;
+  }
+
+  bool operator!=(const Date& other) const {
+    return days_ != other.days_;
+  }
+
+  bool operator<(const Date& other) const {
+    return days_ < other.days_;
+  }
+
+  bool operator<=(const Date& other) const {
+    return days_ <= other.days_;
+  }
+
+  bool operator>(const Date& other) const {
+    return days_ > other.days_;
+  }
+
+  bool operator>=(const Date& other) const {
+    return days_ >= other.days_;
+  }
+
+  // Needed for serialization of FlatVector<Date>
+  operator StringView() const {VELOX_NYI()};
```

Comment:
formatting: move VELOX_NYI on a separate line

Reply:
I tried putting on a separate line. But the format-fix prefers this.